src/network/tarcap/threadpool/priority_queue.cpp

```diff
@@ -0,0 +1,155 @@
+#include "priority_queue.hpp"
+
+namespace taraxa::network::tarcap {
+
+PriorityQueue::PriorityQueue(size_t tp_workers_count, const addr_t& node_addr)
+    : blocked_packets_mask_(), MAX_TOTAL_WORKERS_COUNT(tp_workers_count), act_total_workers_count_(0) {
+  assert(packets_queues_.size() == PacketData::PacketPriority::Count);
+  assert(tp_workers_count >= 1);
+
+  LOG_OBJECTS_CREATE("PRIORITY_QUEUE");
+
+  // high priority packets(consensus - votes) max concurrent workers - 40% of tp_workers_count
+  // mid priority packets(dag/pbft blocks, txs) max concurrent workers - 40% of tp_workers_count
+  // low priority packets(syncing, status, ...) max concurrent workers - 30% of tp_workers_count
+  size_t high_priority_queue_workers = std::max(1, static_cast<int>(MAX_TOTAL_WORKERS_COUNT * 4 / 10));
```

Comment:
I do not understand this, if MAX_TOTAL_WORKERS_COUNT = 10, then we have 4,4,3 so this is more than 10

Reply:
and if we use 9 as max then 3,3,2 so also not good

Reply:
no you dont get the logic here. If MAX_TOTAL_WORKERS_COUNT == 10, there wont be more than 10 threads created. These <>_priority_queue_workers are thread limits for specific priority queues but there is a global limit (MAX_TOTAL_WORKERS_COUNT) for all queues, which has highest priority so it wont be violated. If there is 4 threads in use for high and mid priority queues, for low priority queue will be used only 2 threads, but once 1 thread is from high priority queue is done processing, it can be used in low priority queue as there is max cap 3 threads, etc..